src/regex/engine.cr

```diff
@@ -0,0 +1,4 @@
+require "./pcre"
+
+# :nodoc:
+alias Regex::Engine = PCRE
```

Comment:
Would you consider adding the abstract `Regex::Engine` and `Regex::Engine::MatchData` interfaces in this PR rather than the PR that introduces PCRE2?

Reply:
There is no abstract interface. My idea was that `Regex::Engine` stays just an alias for either `Regex::PCRE` or `Regex::PCRE2`.

Reply:
Fair, I was thinking about extracting the interface [so the engine could be swapped out](https://github.com/crystal-lang/crystal/issues/12790#issuecomment-1331236257), but it makes sense as is if that's a non-goal of this refactor.